components/org.wso2.carbon.identity.application.authenticator.oidc/src/main/java/org/wso2/carbon/identity/application/authenticator/oidc/internal/OpenIDConnectAuthenticatorDataHolder.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright (c) 2017, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.wso2.carbon.identity.application.authenticator.oidc.internal;
+
+import org.wso2.carbon.identity.claim.metadata.mgt.ClaimMetadataManagementService;
+import org.wso2.carbon.user.core.service.RealmService;
+
+public class OpenIDConnectAuthenticatorDataHolder {
+
+    private static OpenIDConnectAuthenticatorDataHolder instance = new OpenIDConnectAuthenticatorDataHolder();
+
+    private RealmService realmService;
+
+    private ClaimMetadataManagementService claimMetadataManagementService;
+
+    private OpenIDConnectAuthenticatorDataHolder() {}
+
+    public static OpenIDConnectAuthenticatorDataHolder getInstance() {
+        return instance;
```

Comment:
What about creating the new instance here.?
if(instance-null)
instance = new OpenIDConnectAuthenticatorDataHolder(); 

Then the static instance will be created when getInstnace() method is invoked.

Reply:
In that case shouldn't we use double locking to make it thread safe??

Reply:
+1 current approach eager loading but we should try to minimize data holders as much as possible

Reply:
the only drawback here is, it will create an instance even if you don't use this instance. But for this case I think this is ok